hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/AbstractFeature.java

```
@@ -95,23 +104,30 @@ protected MirrorTransaction verifyMirrorTransactionsResponse(MirrorNodeClient mi
-            networkTransactionResponse = contractClient.createContract(
-                    fileId,
-                    contractClient
-                            .getSdkClient()
-                            .getAcceptanceTestProperties()
-                            .getFeatureProperties()
-                            .getMaxContractFunctionGas(),
-                    initialBalance == 0 ? null : Hbar.fromTinybars(initialBalance),
-                    null);
-            contractId = verifyCreateContractNetworkResponse();
-            return new DeployedContract(fileId, contractId, compiledSolidityArtifact);
-        }
+    protected synchronized DeployedContract getContract(ContractResource contractResource)
+            throws IOException {
+        return contractIdMap.computeIfAbsent(contractResource, x -> {
+            var resource = resourceLoader.getResource(contractResource.path);
+            try (var in = resource.getInputStream()) {
+                CompiledSolidityArtifact compiledSolidityArtifact = readCompiledArtifact(in);
+                var fileId = persistContractBytes(
+                        compiledSolidityArtifact.getBytecode().replaceFirst("0x", ""));
+                networkTransactionResponse = contractClient.createContract(
+                        fileId,
+                        contractClient
+                                .getSdkClient()
+                                .getAcceptanceTestProperties()
+                                .getFeatureProperties()
+                                .getMaxContractFunctionGas(),
+                        contractResource.initialBalance == 0 ? null : Hbar.fromTinybars(contractResource.initialBalance),
+                        null);
+                ContractId contractId = verifyCreateContractNetworkResponse();
+
+                return new DeployedContract(fileId, contractId, compiledSolidityArtifact);
+            } catch (IOException e) {
+                throw new RuntimeException(e);
```

Might be better to catch all exceptions and log some additional info. For example                 
`log.warn("Issue creating contract: {}, ex: {}", contractResource, e);`

---

Done